donut.c

```diff
@@ -0,0 +1,49 @@
+#include <stdio.h>
+#include <math.h>
+#include <stdint.h>
+#include <string.h>
+#include <unistd.h>
+
+int main() {
+    float A = 0, B = 0;
+    float i, j;
+    int k;
```

Comment:
The variable `z` seems to be used as a 2D array with dimensions 1760 x 4. Consider making this explicit by declaring it as `float z[1760][4]`.